### ISSUE TITLE
Add rule to enforce equals operator alignment

### ISSUE THREAD

**Issue by iandunn** [eslint/eslint#11025]
**Please describe what the rule should do:**

If enabled, the rule should produce a warning when consecutive assignment statements do not have the assignment operator aligned.

This would be similar to using `key-spacing` with `align: colon`.

Optionally, it could be expanded beyond the `=` operator, to include things like aligning the `from` keyword in imports, and similar situations.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Invalid:
```js
const { withSelect } = wp.data;
const { InspectorControls } = wp.editor;
const { Component, Fragment, RawHTML } = wp.element;
```

Valid:
```js
// Single statements would be ignored
const { __ } = wp.i18n;

// Consecutive statements would have to be aligned
const { withSelect }                   = wp.data;
const { InspectorControls }            = wp.editor;
const { Component, Fragment, RawHTML } = wp.element;
```

Invalid:
```js
import classnames from 'classnames';
import { isUndefined, pickBy, split } from 'lodash';
```

Valid:
```js
// Single statements would be ignored
import edit from './edit.jsx';

// Consecutive statements would have to be aligned
import classnames                     from 'classnames';
import { isUndefined, pickBy, split } from 'lodash';
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It passes all of the tests:

* *Widely applicable*: This seems like a common style guideline, but I wasn't able to find any way to enforce it. While searching, I noticed [others](https://stackoverflow.com/questions/52026170/is-there-an-eslint-rule-to-align-on-assignments) trying to achieve [the same thing](https://stackoverflow.com/questions/52718055/eslintrc-json-no-multi-spaces-value-assignment-equal-sign-aligned-javascript) with no solution.
* *Generic*: Yes
* *Atomic*: Yes
* *Unique*: There are some related rules, but none that would produce an error when the operators aren't aligned.
* *Library agnostic*: Yes
* *No conflicts*: There are some related rules (e.g., `no-multi-spaces`), but they can be configured in a way that doesn't produce any errors when the operators are aligned.

**Are you willing to submit a pull request to implement this rule?**

I don't have enough time right now, but possibly in the future.

**Comment by ljharb** [comment]
Are there any common style guides that recommend alignment like this? It seems common in other languages, but the idiomatic JS i've seen recommends strictly avoiding alignment like this.

**Comment by chiqui3d** [comment]
for me is easier to read with alienated, not in zip-zap, so please there is no solution to not show me the errors and I can continue programming with eslint.
